napari/plugins/_builtins.py

```diff
@@ -2,12 +2,21 @@
 Internal napari hook implementations to be registered by the plugin manager
 """
 from pluggy import HookimplMarker
+from typing import Union, List
+from ..types import ReaderFunction, LayerData
+from ..utils import io
 
 napari_hook_implementation = HookimplMarker("napari")
 
 
+def _interal_reader_plugin(path: Union[str, List[str]]) -> List[LayerData]:
+    """Pass ``path`` to our magic_imread function and return as LayerData."""
+    # magic_imread is already capable of accepting both a str or a list
+    return [(io.magic_imread(path),)]
```

Comment:
It seems so close to being able to return `io.imread` directly, rather than this thing... Could we make a decorator that converts `path -> ArrayLike` into `path -> [(ArrayLike,)]` and use that? It seems that more than one package would find this useful...

Reply:
... And I also wonder whether we could somehow also support the `io.magic_read` signature directly anyway... Maybe call it SimpleReaderFunction?

Reply:
sure, love that.

Reply:
I'm ok with that guessing / default behavior, but we might want to keep a close handle on this as we think about easy ways to add meshes, shapes etc too

Reply:
actually, i take that back - see more detailed comment bellow